test/DoctrineBatchUtilsTest/BatchProcessing/SelectBatchIteratorAggregateTest.php

```diff
@@ -115,7 +115,7 @@ public function testIterationWithSuccessfulReFetches(): void
             ['Yadda', ['id' => 123], $freshObjects['foo']],
             ['Yadda', ['id' => 456], $freshObjects['bar']],
         ]);
-        $this->entityManager->expects(self::at(2))->method('clear');
```

Comment:
These `at()` are extremely important, since the execution sequence must be verified. 

Stubbing out the `EntityManager` with a manually crafted `EntityManagerInterface` implementation is preferable.

Something like:

```php
new class implements EntityManagerInterface 
{
    public function (InvocationMocker ...$expectedCall) { ... }
    // ...
}
```

Reply:
Clean solution, I'll do this tomorrow 👍